Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php

```diff
@@ -479,4 +482,86 @@ public function testAbsWorkspaceMappingFilesOutput()
         self::assertCount(0, $components->listConfigurationWorkspaces($options));
     }
+
+    public function testAbsWorkspaceOutputTablesAsFiles()
+    {
+        if (!RUN_SYNAPSE_TESTS) {
+            self::markTestSkipped('Synapse test is disabled.');
+        }
+        $this->clearBuckets();
+        $componentData = [
+            'id' => 'keboola.runner-workspace-abs-test',
+            'data' => [
+                'definition' => [
+                    'type' => 'aws-ecr',
+                    'uri' => '061240556736.dkr.ecr.us-east-1.amazonaws.com/keboola.runner-workspace-test',
+                    'tag' => 'latest',
+                ],
+                'staging_storage' => [
+                    'input' => 'workspace-abs',
+                    'output' => 'workspace-abs',
+                ],
+            ],
+            // https://keboola.slack.com/archives/C02C3GZUS/p1598942156005100
+            // https://github.com/microsoft/msphpsql/issues/400#issuecomment-481722255
+            'features' => ['container-root-user', 'allow-use-file-storage-only'],
+        ];
+
+        $configId = uniqid('runner-test-');
+        $components = new Components($this->client);
+        $configuration = new Configuration();
+        $configuration->setComponentId('keboola.runner-workspace-abs-test');
+        $configuration->setName('runner-tests');
+        $configuration->setConfigurationId($configId);
+        $components->addConfiguration($configuration);
+        $runner = $this->getRunner();
+
+        $runner->run(
+            $this->prepareJobDefinitions(
+                $componentData,
+                $configId,
+                [
+                    'storage' => [
+                        'output' => [
+                            'tables' => [
+                                [
+                                    'source' => 'my-table.csv',
+                                    "destination" => "out.c-synapse-runner-test.test-table",
+                                ],
+                            ],
+                        ],
+                    ],
+                    'parameters' => [
+                        'operation' => 'create-abs-table',
+                    ],
+                    'runtime' => [
+                        'use_file_storage_only' => true,
+                    ],
+                ],
+                []
+            ),
+            'run',
+            'run',
+            '1234567',
+            new NullUsageFile()
+        );
+        // wait for the file to show up in the listing
+        sleep(2);
+
+        // table should not exist
+        self::assertFalse($this->client->tableExists('out.c-synapse-runner-test.test-table'));
+
+        // but the file should exist
+        $fileList = $this->client->listFiles((new ListFilesOptions())->setQuery(
+            'tags:"componentId: keboola.runner-workspace-abs-test" AND tags:' .
+            sprintf('"configurationId: %s"', $configId)
+        ));
```

Comment:
can we assert the file name?

Reply:
oh, @odinuv  this raises a question,  I guess the file name should be $table['destination'] eh?  like in this example is should be `out.c-synapse-runner-test.test-table`

Reply:
i've no idea what it is :) and afaik there is no requirement for the filename, so anything is fine with me. I just want to make a status-quo out of it.